### ISSUE TITLE
Default event not prevented in browsers lacking event.preventDefault

### ISSUE THREAD

**Issue by polm** [fabien-d/alertify.js#104]
I've only confirmed this with IE9, though I imagine it applies in older IE versions as well.

There's a item-specific screen in my application with the following expected flow:
1. Click button to start processing (firing Backbone.js route)
2. Alertify.js pops up a confirmation (`alertify.confirm`)
3. If "OK", a loading gif is shown
4. When processing finishes, a results page is shown

I ran into trouble in IE because the hash link (`href="#"`) is being followed, which means the flow looks like this:
1. Click button
2. Confirm box
3. If OK, **fire default Backbone.js route and return to top page** (this is the problem)
4. When processing finishes, show the results page

I noticed that you're using `preventDefault` if it exists, but it doesn't in IE9. I was able to get the expected behavior by adding the following line to my click handler:

```
if (event) event.returnValue = false;
```

**Comment by fabien-d** [comment]
What version of alertify are you using? I fixed a similar issue at `0.3.3` so it may simply be an outdated version.

Also, could you double check the behaviour on the [demo page](http://fabien-d.github.com/alertify.js/) (in IE9) to see if you can reproduce the same functionality (I wasn't able to)? I added a return false at the bottom of the function to prevent the default event from firing for browsers that don't support `event.preventDefault`. If a piece of code in your callback is causing an error, the return false would never be reached and the default event would be fired.

I will look into the `returnValue` property... so instead of just testing against `event.preventDefault`... do something along the lines of `event.preventDefault ? event.preventDefault() : event.returnValue = false;` to try and prevent the default event from ever firing.

**Comment by polm** [comment]
Thanks for your reply; I'm not at the machine with the code right now, but I did check to make sure that it wasn't happening due to an exception. On Monday I'll confirm the version and make something reproducible. 

One thing I noticed reading documentation is that in IE9 events do have a preventDefault method, but only if you have a `strict` doctype. Whether or not they have the preventDefault, they don't always respect a `return false` (see [search results](https://www.google.co.jp/search?q=ie+ignores+return+false)). 

I like your code, but note that `event` there is `window.event`, which is not necessarily the argument to the onClick handler.
